addons/mail/static/src/core/common/thread_model.js

```diff
@@ -50,7 +50,7 @@ export class Thread extends Record {
                 request_list: fieldNames,
             });
             thread = this.get(data);
-            if (!thread.exists() || !thread.hasReadAccess) {
+            if (!thread?.exists()) {
```

Comment:
Could we test the read access after the check on thread? So we can keep the hasReadAccess?
```js
            if (!thread?.exists()) {
                return;
            } else if (!thread.hasReadAccess) {
                return;
            }
```

Reply:
But we don't want to keep it. It's just useless and inconsistent code.

If the same method is called a second time in a row with the exact same params the thread will be returned immediately regardless of `hasReadAccess` as the line 1 of the function will find it.

Reply:
Well that make sens